src/Perl6/Metamodel/RolePunning.nqp

```diff
@@ -57,6 +57,24 @@ role Perl6::Metamodel::RolePunning {
         self.pun($obj)
     }
 
+    my $proxy_type := nqp::null();
+    method !make_proxy($cont) {
+        if nqp::isnull($proxy_type) {
+            $proxy_type := nqp::gethllsym('Raku', 'Proxy');
+        }
+
+        my $proxy   := nqp::create($proxy_type);
```

Comment:
Why not a construct with `nqp::ifnull($proxy_type,nqp::gethllsym('Raku', 'Proxy'))` ?

Reply:
Does that bind `$proxy_type` to the `Proxy` type object?

Reply:
Good point.  No.  I actually meant:

````raku
nqp::ifnull($proxy_type,$prxy_type := nqp::gethllsym('Raku', 'Proxy'))
````

Reply:
I guess that would be a good micro-optimization. But not as good as eager initialization, which appears to be possible after all.

Reply:
Indeed...  But I figured that wasn't possible, as this is a precursor to the Bootstrap, is it not?  And `Proxy` only gets defined in the bootstrap I believe?

Reply:
It appears that the Bootstrap is compiled before RolePunning. Both `nmake` and `nmake test` ran fine. But then `nmake spectest` hung, so I aborted:

```
t\spec\S17-promise\lock-async.t ................................... ok
===(   66393;538  21/28  0/?  0/?  0/?  0/?  0/? )======================NMAKE : fatal error U1058: terminated by user Stop.
```

I've seen that test hanging before, so I don't think it's related. I'll try again.

Reply:
Eager initialization causes spectests to fail. I'll go for the micro-optimization instead.

Reply:
On second thought, it doesn't feel right to use `nqp::ifnull` without using the return value. And I don't like to optimize without measuring, which I consider too much work for this case.

Please feel free to apply the optimization yourself if you think it's worth it :)

Reply:
`RolePunning` cannot compile after the bootstrap because it is in the Metamodel.